### DISCUSSION TITLE
Missing commits on Release notes

### DISCUSSION THREAD

**Discussion by lucasssfigueredo** [rfennell/AzurePipelines#1126] [Q&A] [answered]
Hi all.

I have the Generate Release Notes task added to my Release Pipeline which has 94 commits for the first execution of this pipeline:
![image](https://user-images.githubusercontent.com/18660314/140093060-a2fc9cfe-3326-44b6-aa6d-c7ec128c45d9.png)

For some reason, the Build that triggered this Release is showing 50 commits associated, and that's exactly the number of commits I see on the Release Notes.
![image](https://user-images.githubusercontent.com/18660314/140093683-bf643dd5-c0b0-4781-b1d1-4411cb4e0067.png)

I had the idea that once I am running the tool from a Release Pipeline, it would bring the Commits from the Release and not from the Build. 

Any ideas?

**Comment by rfennell** [comment]
No immediate idea, if you could run the pipeline with the variable `system.debug=true` so we get a detailed log of the calls made we might spot something. Maybe a call is not set to get all items, just the first 50.

You might find the [command line tester](https://github.com/rfennell/AzurePipelines/tree/main/Extensions/XplatGenerateReleaseNotes/V3/testconsole) useful to allow repeated running of the generation process without rerunning the task (this defaults to detailed logging buy the way)

**Reply by lucasssfigueredo** [reply]
Hi Richard, re-executed the release pipeline with payload to console
[tasklog_9.log](https://github.com/rfennell/AzurePipelines/files/7475516/tasklog_9.log)

We have been using the tool for some time now, but this was the first time with a big number of commits. Maybe there is a limit set?

**Reply by rfennell** [reply]
I have just created a PR #1132 there was an issue that on the first run against a release it would only get the default (50) items, were the other calls all used the maximum of 5000.

If you want to try this branch with the command line testers to prove it work. If does I will issue a new release with the fix.

**Reply by lucasssfigueredo** [reply]
Hi Richard, 

Thanks a lot for your prompt response. I need to learn how to test it locally :). I'll keep you posted. 

Thank you.

**Reply by rfennell** [reply]
If you testing it with the test harness is too much of a problem, I can just publish the change. It has already been through my automated tests and passed.

**Reply by lucasssfigueredo** [reply]
Please, go ahead. I will re-run the Release as soon as I hear from you.

**Reply by rfennell** [reply]
Has been released as 3.73.2

**Reply by lucasssfigueredo** [reply]
It worked now 👍 

Thanks again

**Comment by rfennell** [comment, marked as answer]
Fixed in 3.73.2
